frontend_tests/node_tests/compose_validate.js

```diff
@@ -711,93 +719,39 @@ test_ui("warn_if_mentioning_unsubscribed_user", ({override, override_rewire, moc
     };
 
     // Test with invalid stream in compose box. It should return noop.
-    compose_validate.warn_if_mentioning_unsubscribed_user(mentioned);
-    assert.equal($("#compose_invite_users").visible(), false);
+    new_banner_rendered = false;
+    compose_validate.warn_if_mentioning_unsubscribed_user(mentioned_user);
+    assert.ok(!new_banner_rendered);
 
     // Test mentioning a user that should gets a warning.
-
-    const checks = [
-        (function () {
-            let called;
-            override_rewire(compose_validate, "needs_subscribe_warning", (user_id, stream_id) => {
-                called = true;
-                assert.equal(user_id, 34);
-                assert.equal(stream_id, 111);
-                return true;
-            });
-            return function () {
-                assert.ok(called);
-            };
-        })(),
-
-        (function () {
-            let called;
-            mock_template("compose_invite_users.hbs", false, (context) => {
-                called = true;
-                assert.equal(context.user_id, 34);
-                assert.equal(context.stream_id, 111);
-                assert.equal(context.name, "Foo Barson");
-                return "fake-compose-invite-user-template";
-            });
-            return function () {
-                assert.ok(called);
-            };
-        })(),
-
-        (function () {
-            let called;
-            $("#compose_invite_users").append = (html) => {
-                called = true;
-                assert.equal(html, "fake-compose-invite-user-template");
-            };
-            return function () {
-                assert.ok(called);
-            };
-        })(),
-    ];
-
-    mentioned = {
+    mentioned_user = {
         email: "foo@bar.com",
         user_id: 34,
         full_name: "Foo Barson",
     };
+    people.add_active_user(mentioned_user);
 
     stream_data.add_sub(sub);
-    compose_validate.warn_if_mentioning_unsubscribed_user(mentioned);
-    assert.equal($("#compose_invite_users").visible(), true);
-
-    for (const f of checks) {
-        f();
-    }
+    new_banner_rendered = false;
+    $("#compose_banners .recipient_not_subscribed").length = 0;
+    compose_validate.warn_if_mentioning_unsubscribed_user(mentioned_user);
+    assert.ok(new_banner_rendered);
```

Comment:
This test is no longer validating any of the details of the warning; just a boolean for whether something was rendered. 

I think the right structure for this class of test would be to just verify an exact object we expect to have been passed to the template; I guess the old one did that in some form. I'm not sure it's worth addressing in this PR.

Reply:
It tests the details of the warning in the template mock, in the way you're describing, above:

```
    mock_template("compose_banner/not_subscribed_warning.hbs", false, (data) => {
        assert.equal(data.classname, compose_error.CLASSNAMES.recipient_not_subscribed);
        assert.equal(data.user_id, 34);
        assert.equal(data.stream_id, 111);
        assert.equal(data.name, "Foo Barson");
        new_banner_rendered = true;
    });
```

Reply:
OK sounds good; had missed that because a lot of similar looking things were removed; this test was pretty messy :).

Reply:
hopefully much cleaner now!